tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/BuilderUtil.scala

```diff
@@ -26,9 +26,60 @@ object BuilderUtil {
   /** Removes elem from the scope, as the scope only contains types of free variables */
   def markAsBound(elem: TlaEx): TBuilderInternalState[Unit] = State[TBuilderContext, Unit] { mi: TBuilderContext =>
     require(elem.isInstanceOf[NameEx])
-    (mi.copy(nameScope = mi.nameScope - elem.asInstanceOf[NameEx].name), ())
+    (mi.copy(freeNameScope = mi.freeNameScope - elem.asInstanceOf[NameEx].name), ())
   }
 
+  /** Performs shadowing checks for ternary operators which introduce bound variables */
+  def boundVarIntroductionTernary(
+      rawMethod: (TlaEx, TlaEx, TlaEx) => TlaEx // order: variable, set, expression
```

Comment:
I don't think I understand the punctuation in the comment, and I'm not sure I understand the comment either. Doe each word relate to one of the `TlaEx` in the type what is an "order" expression?

Reply:
What are "methods" in this context?

Reply:
`_choose`, `_forall`, `_exists`

Reply:
What are those? TLA operators? Scala functions?